src/fe/fe_base.C

```diff
@@ -744,25 +744,25 @@ void FEGenericBase<Real>::compute_dual_shape_coeffs ()
   dual_coeff.resize(sz, sz);
   DenseMatrix<Real> A(sz, sz), D(sz, sz);
 
-  const std::vector<Real> JxW = this->get_JxW();
-
-  for (auto i : index_range(phi))
-    for (auto qp : index_range(phi[i]))
+  const std::vector<Real> weights = qrule.get_weights();
+  // we do not need J here as it will be canceled in the A^-1*D calculation anyways
+  for (auto i : index_range(dual_phi))
+    for (auto qp : index_range(dual_phi[i]))
     {
-      D(i,i) += JxW[qp]*phi[i][qp];
-      for (auto j : index_range(phi))
-        A(i,j) += JxW[qp]*phi[i][qp]*phi[j][qp];
+      D(i,i) += weights[qp]*dual_phi[i][qp];
+      for (auto j : index_range(dual_phi))
```

Comment:
Same here and for all following. You never know, someone nefarious like @jwpeterson might come along and try to change your index when you least expect it :laughing: